directus-remix/index.md

```diff
@@ -0,0 +1,331 @@
+
+```ts
+import { createDirectus, rest, } from '@directus/sdk';
+
+//Structure of the data you will be fetching from Directus
+type Global = {
+  title: string;
+  description: string;
+}
+type Author = {
+  name: string
+}
+type Page = {
+  title: string;
+  content: string;
+  slug: string;
+}
+type Post = {
+  image: string;
+  title: string;
+  author: Author;
+  content: string;
+  published_date: string
+  slug: string;
+}
+
+type Schema = {
+  posts: Post[];
+  global: Global;
+  pages: Page[];
+}
+
+const directus = createDirectus<Schema>(YOUR_DIRECTUS_URL).with(rest());
+
+export default directus;
+
+
+```
+
+Ensure your Directus URL is correct when initializing the Directus JavaScript SDK.
+
+## Using Global Metadata and Settings
+
+In your Directus project, navigate to Settings -> Data Model and create a new collection called `global`. Under the Singleton option, select 'Treat as a single object', as this collection will have just a single entry containing global website metadata.
+
+Create two text input fields - one with the key of `title` and one `description`.
+
+Navigate to the content module and enter the global collection. Collections will generally display a list of items, but as a singleton, it will launch directly into the one-item form. Enter information in the title and description field and hit save.
+
+ ![A form named "Global" has two inputs - a title and a description, each filled with some text.](global-data.png)
+
+ By default, new collections are not accessible to the public. Navigate to Settings -> Roles & Permissions -> Public and give Read access to the Global collection.
+
+ In your `app/routes/_index.tsx` file, add the following to fetch the data from Directus and display it:
+
+```tsx
+import directus from "../lib/directus";
+import { readSingleton } from "@directus/sdk";
+import { useLoaderData } from "@remix-run/react";
+
+
+export const loader = async () => {
+  const global = await directus.request(readSingleton("global"));
+  return global
+};
+
+export default function Index() {
+  const { title, description } = useLoaderData< typeof loader>();
+  return (
+    <div>
+      <h1>{title}</h1>
+      <p>{description}</p>
+    </div>
+  );
+}
+
+```
+
+Refresh your browser. You should see the data from your Directus Global collection displayed in the homepage.
+
+## Creating Pages With Directus
+
+### Set up Directus
+
+Create a new collection called `pages` - make the Primary ID Field a "Manually Entered String" called `slug`, which will correlate with the URL for the page. For example `about` will later correlate to the page `localhost:3000/about`.
+
+Create a text input field called `title` and a WYSIWYG input field called `content`. In Roles & Permissions, give the Public role read access to the new collection.
+
+Create some items in the new collection
+
+### Setup Dynamic Routes in Astro
```

Comment:
I know you're dong a few of these, but please take some time to ensure copy/paste work is then adjusted appropriately!